server/server.go

```diff
@@ -103,6 +103,7 @@ func NewServer(c *Config) (*Server, error) {
 		server.reverseProxy = httputil.NewSingleHostReverseProxy(u)
 		//always use proxy host
 		server.reverseProxy.Director = func(r *http.Request) {
+			r.Header.Add("X-Forwarded-Host", r.Host)
```

Comment:
this "leaks" information, it tells the backend server where the "frontend" is - probably not an issue in 95% of cases - though i think this needs to be user defined. maybe you could wrap with `if EnvBool("X_FORWARDED_HOST") { ... }` ?